test/acs-engine-test/main.go

```diff
@@ -69,79 +75,97 @@ func (m *TestManager) Run() error {
 		go func(i int, d Deployment) {
 			defer m.wg.Done()
 
-			name := strings.TrimSuffix(d.ClusterDefinition, filepath.Ext(d.ClusterDefinition))
-			instanceName := fmt.Sprintf("acse-%d-%s-%s-%d", rand.Intn(0x0ffffff), d.Location, os.Getenv("BUILD_NUMBER"), i)
-			resourceGroup := fmt.Sprintf("test-acs-%s-%s-%s-%d", strings.Replace(name, "/", "-", -1), d.Location, os.Getenv("BUILD_NUMBER"), i)
-			logFile := fmt.Sprintf("%s/%s.log", logDir, resourceGroup)
-
-			// determine orchestrator
-			env := os.Environ()
-			env = append(env, fmt.Sprintf("CLUSTER_DEFINITION=examples/%s", d.ClusterDefinition))
-			cmd := exec.Command("test/step.sh", "get_orchestrator_type")
-			cmd.Env = env
-			out, err := cmd.Output()
-			if err != nil {
-				wrileLog(logFile, "Error [getOrchestrator %s] : %v", d.ClusterDefinition, err)
-				retvals[i] = 1
-				return
-			}
-			orchestrator := strings.TrimSpace(string(out))
-
-			// update environment
-			env = append(env, fmt.Sprintf("LOCATION=%s", d.Location))
-			env = append(env, fmt.Sprintf("ORCHESTRATOR=%s", orchestrator))
-			env = append(env, fmt.Sprintf("INSTANCE_NAME=%s", instanceName))
-			env = append(env, fmt.Sprintf("DEPLOYMENT_NAME=%s", instanceName))
-			env = append(env, fmt.Sprintf("RESOURCE_GROUP=%s", resourceGroup))
-
-			steps := []string{"generate_template", "deploy_template"}
-
-			// determine validation script
-			if !d.SkipValidation {
-				validate := fmt.Sprintf("test/cluster-tests/%s/test.sh", orchestrator)
-				if _, err = os.Stat(fmt.Sprintf("%s/%s", m.rootDir, validate)); err == nil {
-					env = append(env, fmt.Sprintf("VALIDATE=%s", validate))
-					steps = append(steps, "validate")
-				}
-			}
-			for _, step := range steps {
-				txt, err := runStep(resourceGroup, step, m.rootDir, env, timeout)
+			for attempt := 0; attempt < retries; attempt++ {
+				name := strings.TrimSuffix(d.ClusterDefinition, filepath.Ext(d.ClusterDefinition))
+				instanceName := fmt.Sprintf("acse-%d-%s-%s-%d-%d", rand.Intn(0x0ffffff), d.Location, os.Getenv("BUILD_NUMBER"), i, attempt)
+				resourceGroup := fmt.Sprintf("x-%s-%s-%s-%d-%d", strings.Replace(name, "/", "-", -1), d.Location, os.Getenv("BUILD_NUMBER"), i, attempt)
+				logFile := fmt.Sprintf("%s/%s.log", logDir, resourceGroup)
+				var validateLogFile string
+
+				// clear the return value
```

Comment:
The if depth in this if statement gets pretty deep and is making the PR harder then needed. Can you make a method for the loop we are retrying and possibly one for the loop it self so you can call go method(i,d) This is getting a little long today